src/experiment_generator/tmp_parser/yaml_config.py

```diff
@@ -1,29 +1,41 @@
 """
 A temporary parser for YAML format files.
-
 - `read_yaml`: Load a YAML file into a Python dictionary with preserved quotes.
 - `write_yaml`: Dump a Python dictionary back to a YAML file, maintaining format.
 """
 
+import io
+import re
 import ruamel.yaml
 
 ryaml = ruamel.yaml.YAML()
 ryaml.indent(mapping=2, sequence=4, offset=2)
-
 ryaml.preserve_quotes = True
-
-
-def read_yaml(yaml_path: str) -> dict:
-    """
-    Reads a YAML file and returns a dictionary.
-    """
-    with open(yaml_path, "r", encoding="utf-8") as f:
+ryaml.width = 10**9  # disable line wrapping for long lines
+
+# remove blank lines between keys:
+#   key1: ...
+#
+#   key2: ...
+# ref: https://www.regextester.com/pregsyntax.html
+_BLANK_BETWEEN_KEYS = re.compile(
+    r"(?m)"  # multi-line mode and don't stop the regex on a line break
+    r"^(?P<i>[ \t]*)(?P<k>[^ \t#\-\n][^:\n]*:.*)\n"  # a key line
+    r"(?:[ \t]*\n)+"  # one+ blank lines
+    r"(?=(?P=i)[^ \t#\-\n][^:\n]*:)"  # next line is a key at same indent
+)
+
+
+def read_yaml(path: str):
```

Comment:
Perhaps should keep the dosctring?
```suggestion
def read_yaml(path: str):
    """
    Reads a YAML file and returns a dictionary.
    """
```